File: server/spring-boot-ecommerce/src/main/java/com/ecommerce/springbootecommerce/dto/order/OrderRequest.java
package com.ecommerce.springbootecommerce.dto.order;

import com.ecommerce.springbootecommerce.dto.address.AddressDTO;
import com.ecommerce.springbootecommerce.dto.user.UserDTO;
import lombok.Data;

import javax.validation.constraints.NotNull;
import java.util.List;

@Data
public class OrderRequest {

    @NotNull(message = "user cannot be empty")
    private UserDTO user;

    @NotNull(message = "shipping address cannot be empty")
    private AddressDTO shippingAddress;

    @NotNull(message = "order cannot be empty")
    private OrderDTO order;

    @NotNull(message = "order items cannot be empty")
    private List<OrderItemDTO> orderItems;
}

File: client/ecommerce-app/src/app/cart/models/orderRequest.ts
import { Address } from './address';
import { Order } from './order';
import { OrderItem } from './orderItem';
import { User } from '../../profile/models/user';

export class OrderRequest {
  user!: User;
  shippingAddress!: Address;
  order!: Order;
  orderItems!: OrderItem[];
}
